src/plugins/default/EventExclusionPlugin.ts

```diff
@@ -0,0 +1,86 @@
+import { IInnerError } from '../../models/IInnerError';
+import { IEventPlugin } from '../IEventPlugin';
+import { EventPluginContext } from '../EventPluginContext';
+import { Utils } from '../../Utils';
+
+export class EventExclusionPlugin implements IEventPlugin {
+  public priority: number = 45;
+  public name: string = 'EventExclusionPlugin';
+
+  public run(context: EventPluginContext, next?: () => void): void {
+    function getLogLevel(level: string): number {
+      switch (level.toLowerCase()) {
+        case 'trace':
+          return 0;
+        case 'debug':
+          return 1;
+        case 'info':
+          return 2;
+        case 'warn':
+          return 3;
+        case 'error':
+          return 4;
+        case 'fatal':
+          return 5;
+        case 'off':
+          return 6;
+        default:
+          return -1;
+      }
+    }
+
+    function getMinLogLevel(settings: Object, loggerName: string = '*'): number {
+      return getLogLevel(getTypeAndSourceSetting(settings, 'log', loggerName, 'Trace'));
+    }
+
+    function getTypeAndSourceSetting(settings: Object = {}, type: string, source: string, defaultValue: string|boolean = undefined): string|boolean {
+      if (!type) {
+        return defaultValue;
+      }
+
+      let sourcePrefix =  `@@${type}:`;
+      if (settings[sourcePrefix + source]) {
+        return settings[sourcePrefix + source];
+      }
+
+      // check for wildcard match
+      for (let key in settings) {
+        if (Utils.startsWith(key.toLowerCase(), sourcePrefix.toLowerCase()) && Utils.isMatch(source, [key.substring(sourcePrefix.length)])) {
```

Comment:
Doesn't seem like this would handle the same wildcard patterns as the .NET client. In the .NET client the pattern can start with an * which will match based on suffix, end with a * which will match based on prefix, start and end with a * which will do a substring match or it can just be a * which will match everything.